README.md

```diff
@@ -64,7 +64,7 @@ role_arn   | Optional. The AWS role to assume. Useful for retrieving cross accou
 metrics  | Required. A list of CloudWatch metrics to retrieve and export
 aws_namespace  | Required. Namespace of the CloudWatch metric.
 aws_metric_name  | Required. Metric name of the CloudWatch metric.
-aws_dimensions | Optional. Which dimension to fan out over.
+aws_dimensions | Required. This should contain exactly all the dimensions available for a metric. Run `aws cloudwatch list-metrics` to find out which dimensions you need to include for your metric.
```

Comment:
This is incorrect, the user must chose which ones they want and this can be empty.

Reply:
Are you sure? I couldn't get it to work with this example config:

```
config: |-
  region: us-west-2
  metrics:
   - aws_namespace: AWS/Kafka
     aws_metric_name: SumOffsetLag
```

What is the expected behavior when `aws_dimensions` is empty?

Only after adding these exact dimensions:
```
aws_dimensions: [Consumer Group, Cluster Name, Topic]
```

was I able to retrieve any metrics. Any other combination did not work for me.

Reply:
Yes, I'm sure. When it's empty, you should get a fully aggregated result. Unfortunately Cloudwatch at times doesn't follow its own data model for certain services, and will instead return nothing for certain combinations of dimensions.

Reply:
Well that's unfortunate then because it does not work for the metric that I mentioned. Which means that the `aws_dimensions` is not optional in all cases.

While my update to the docs may not be technically correct, it does suggest a configuration that is guaranteed to work.

I would suggest adding to the documentation that `aws_dimensions` is not optional for all metrics. With the addition to try adding all the dimensions from `aws cloudwatch list-metrics` in case nothing is returned. That would help future users like myself save a lot of time and trial & error while setting this up.

Reply:
> While my update to the docs may not be technically correct, it does suggest a configuration that is guaranteed to work.

I'd not presume that unfortunately.

Reply:
*shakes fist at computers* thank you @brian-brazil for setting us straight

@agarella I agree that we should document this fully and help users as much as possible. Do you want to take a stab at this? I think it deserves a whole section with a header so that it stands out.